README.md

```diff
@@ -1,2 +1,246 @@
+import com.bobocode.hoverla.bring.context.ApplicationContext;
+
+public class DemoApp {
+    public static void main(String[] args) {
+        ApplicationContext = BringApplication.getContextBuilder()
+                .logLevel(Level.DEBUG)                 // provide logging level
+                .packagesToScan("package.to.scan")     // provide packages to be scanned for beans
+                .build();
+    }
+}
+```
+
+### Bean
+
+---
+**Bean** is a class that is managed by IoC container. To define a bean you have multiple options:
+
+- [Annotation Configuration](#annotation-configuration)
+- [Java Configuration](#java-configuration)
+
+### Annotation configuration
+
+---
+To define a bean with **annotation configuration** just annotate your desired bean class with
+[@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java) annotation. Look at the next example:
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+
+@Bean
+public class MySimpleBean {
+
+}
+```
+
+By default, **Bring will use a class name as a bean name**. You can customize it by providing a `name` property for
+the [@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java) annotation
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+
+@Bean("customBean")
+public class MyCustomBean() {
+
+}
+```
+
+If your class has dependencies that should be injected, you may use one of these types of injection:
+
+- [Constructor Injection](#constructor-injection)
+- [Field Injection](#field-injection)
+
+A bean class has the following constraints:
+
+- only one constructor with fields should be
+  present. [@Inject](src/main/java/com/bobocode/hoverla/bring/annotation/Inject.java) annotation may be used, although
+  it's not necessary
+- use only [constructor injection](#constructor-injection) or [field injection](#field-injection).
+  Using of both injection types **is not recommended**, because it may lead to unpredictable results
+
+### Java configuration
+
+---
+With Java Configuration you can provide a way how a bean should be instantiated via Java code. Use
+a [@Configuration](src/main/java/com/bobocode/hoverla/bring/annotation/Configuration.java)
+annotation to mark a class that will contain bean declarations. To declare a bean create a method and mark it
+as [@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java)
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+import com.bobocode.hoverla.bring.annotation.Configuration;
+
+@Configuration
+public class MyJavaConfiguration {
+
+    @Bean
+    public MyBean myBean() {
+        return new MyBean();
+    }
+}
+```
+
+By default, **Bring will use a method name as a bean name**. The bean name can be provided via
+the [@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java) annotation
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+
+public class MyJavaConfiguration {
+
+    @Bean("customBean")
```

Comment:
Bean(name = "customBean")
We don't pass it as annotation value

Reply:
fixed